### PR TITLE
TDD project

### PR DESCRIPTION
**Pull request by svitalis123**
# TDD project
## I accomplished

- [ ] Creating tests
- [ ] Writing good enough code
- [ ] refractor my code

### CODE REVIEW

File: spec/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+require_relative '../solver'
+describe Solver do
+  context 'When passed an input' do
+    solver = Solver.new
+    it 'expect factorial of 5 tobe 120' do
+      expect(solver.factorial(5)).to eq 120
+    end
+    it 'expect positive number ' do
+      expect(solver.factorial(5)).to be >= 0
+    end
```
Comment (FrederickMih) [inline spec/solver_spec.rb:10]: - [ ] Based on the requirements of this project, the #factorial method should have a test that handles negative value which raises an exception as an expectation. Kindly fix :wrench: this.

Comment (FrederickMih) [inline spec/solver_spec.rb:10]: - [ ] Please also make sure to handle the special cases of the #factorial method - when input is 0, the result should equal 1

File: spec/solver_spec.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    it 'Expect hello to be reversed to olleh' do
+      expect(solver.reverse('hello')).to eq 'olleh'
+    end
+    it 'Expect 6 to return fizz' do
+      expect(solver.fizzbuzz(6)).to eq 'fizz'
+    end
+    it 'Expect 10 to return buzz' do
```
Comment (FrederickMih) [inline spec/solver_spec.rb:23]: - [ ] I think there's another scenario left. Now let's kindly use a  number(N) that's not divided by 6 or 10. The result should be equal to that number that's N

**Review by FrederickMih — CHANGES_REQUESTED** [review]
## Hi :raising_hand:  @svitalis123, @samuelmoiwa , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

Screenshot from 2022-10-20 09-38-51



## Highlights :green_circle: 

- Professional PR :heavy_check_mark: 
- Good commit history :+1: 


## Required Changes ♻️

- [ ] The Microverse logo on the top of the README.md file should be removed. The text "Microverse README Template" should also be removed.
- [ ] The License link in the README file is broken. Please let's fix the link to render the readme professionally:+1: 
- [ ] Please kindly customized your README file to match this project. All the content should match this project and not the school library app.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by sja-thedude — CHANGES_REQUESTED** [review]
## Required Changes ♻️

Hi @svitalis123, This is @sja-thedude your code reviewer.

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

200w

 ## Highlights
✔️ Good job with making changes
✔️ All linters passing

 ## Required Changes ♻️
Ass suggested by the previous reviewer, Kindly make sure to make changes in the readme given below:
- [ ] The Microverse logo on the top of the README.md file should be removed. The text "Microverse README Template" should also be removed.
- [ ] The License link in the README file is broken. Please let's fix the link to render the readme professionally/
- [ ] Please kindly customized your README file to match this project. All the content should match this project and not the school library app.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Ghiftee — APPROVED** [review]
## STATUS: APPROVED ♻️♻️
Hi @svitalis123 and @samuelmoiwa, 

To highlight 👍🏽 
✔️ Your readme is professional
✔️ Your tests pass

Your project is complete!  There is nothing else to say other than... it's time to merge it 🚀  
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Pull request merged**